test/e2e/steps/steps.go

```diff
@@ -0,0 +1,531 @@
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	olmNamespace      = "olmv1-system"
+	olmDeploymentName = "operator-controller-controller-manager"
+	timeout           = 300 * time.Second
+	tick              = 1 * time.Second
+)
+
+var kubeconfigPath string
+
+func RegisterSteps(sc *godog.ScenarioContext) {
+	sc.Step(`^OLM is available$`, OLMisAvailable)
+	sc.Step(`^bundle "([^"]+)" is installed in version "([^"]+)"$`, BundleInstalled)
+	sc.Step(`^ClusterExtension is applied$`, ResourceIsApplied)
+	sc.Step(`^ClusterExtension is updated$`, ResourceIsApplied)
+	sc.Step(`^ClusterExtension is available$`, ClusterExtensionIsAvailable)
+	sc.Step(`^ClusterExtension is rolled out$`, ClusterExtensionIsRolledOut)
+	sc.Step(`^ClusterExtension reports ([[:alnum:]]+) as ([[:alnum:]]+) with Reason ([[:alnum:]]+):$`, ClusterExtensionReportsCondition)
+	sc.Step(`^ClusterExtension reports ([[:alnum:]]+) as ([[:alnum:]]+) with Reason ([[:alnum:]]+)$`, ClusterExtensionReportsConditionWithoutMsg)
+	sc.Step(`^ClusterExtension reports ([[:alnum:]]+) as ([[:alnum:]]+)$`, ClusterExtensionReportsConditionWithoutReason)
+	sc.Step(`^resource "([^"]+)" is installed$`, ResourceAvailable)
+	sc.Step(`^resource "([^"]+)" is available$`, ResourceAvailable)
+	sc.Step(`^resource "([^"]+)" is removed$`, ResourceRemoved)
+	sc.Step(`^resource is applied$`, ResourceIsApplied)
+	sc.Step(`^resource apply fails with error msg containing "([^"]+)"$`, ResourceApplyFails)
+	sc.Step(`^resource "([^"]+)" is eventually restored$`, ResourceRestored)
+	sc.Step(`^resource "([^"]+)" matches$`, ResourceMatches)
+	sc.Step(`^Service account "([^"]*)" with needed permissions is available in test namespace$`, ServiceAccountWithNeededPermissionsIsAvailableInNamespace)
+	sc.Step(`^Service account "([^"]*)" in test namespace is cluster admin$`, ServiceAccountWithClusterAdminPermissionsIsAvailableInNamespace)
+	sc.Step(`^Service account "([^"]+)" in test namespace has permissions to fetch "([^"]+)" metrics$`, ServiceAccountWithFetchMetricsPermissions)
+	sc.Step(`^Service account "([^"]+)" sends request to "([^"]+)" endpoint of "([^"]+)" service$`, SendMetricsRequest)
+	sc.Step(`^"([^"]+)" catalog is updated to version "([^"]+)"$`, CatalogIsUpdatedToVersion)
+	sc.Step(`^"([^"]+)" catalog serves bundles$`, CatalogServesBundles)
+	sc.Step(`^"([^"]+)" catalog image version "([^"]+)" is also tagged as "([^"]+)"$`, TagCatalogImage)
+	sc.Step(`^operator "([^"]+)" target namespace is "([^"]+)"$`, OperatorTargetNamespace)
+	sc.Step(`^Prometheus metrics are returned in the response$`, PrometheusMetricsAreReturned)
+}
+
+func init() {
+	kubeconfigPath = os.Getenv("HOME") + "/.kube/config"
+}
+
+var (
+	logger = log.Log
+)
+
+func kubectl(args ...string) (string, error) {
+	cmd := exec.Command("kubectl", args...)
```

Comment:
Since we run these tests in other environments, we might need to consider the use of `oc` as a substitute for `kubectl`. Here and elsewhere.

Reply:
added `--k8s.cli` command arg that can tweak that.